lib/bloc_record/selection.rb

```diff
@@ -33,6 +38,65 @@ def find_by(attribute, value)
     init_object_from_row(row)
   end
 
+  def find_each(options = {}, &block)
```

Comment:
Adding `block` does not solve the original problem (and is not necessary).

If I just call `find_each()`, then `options` becomes `{}`, not `nil`. so it will fall into your `else` producing the line `LIMIT  OFFSET  ;`, and that will explode due to invalid syntax. There are a number of cases that could occur that you have to make sure does not produce invalid SQL: find_each, `find_each(start: 20)`, `find_each(batch_size: 20)`, `find_each(start: 20, batch_size: 20)`, `find_each(some_key: "trollolololol")` (whether you ignore extra keys or throw an exception is up to you).